examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+  
+Once you have created the `DataProvider` objects above, you are ready to
+initialize the Casting App as described below. Note: When you initialize the
+Casting client, make sure your code initializes it only once, before it starts a
+Matter casting session.
+
+On Android, create an AppParameters object using the
+`rotatingDeviceIdUniqueIdProvider`, `commissioningDataProvider` and
+`dacProvider`, and call `CastingApp.initialize` with it.
+
+```java
+public static void demoInitialization() {
+    // Create an AppParameters object to pass in global casting parameters to the SDK
+    final AppParameters appParameters = new AppParameters(
+        rotatingDeviceIdUniqueIdProvider,
+        commissioningDataProvider,
+        dacProvider);
+
+    // Initialize the SDK using the appParameters and check if it returns successfully
+    MatterError err = CastingApp.initialize(appParameters);
+    if(err.hasError()) {
+        Log.e(TAG, "Failed to initialize Matter CastingApp");
+    }
+}
+```
+
+On iOS, call `MTRCastingApp.initialize` with an object of the
+`AppParametersDataSource`.
+
+```objectivec
+init()
+{
+    ...
+    MTRCastingApp.initialize(with: AppParametersDataSource())
+}
+```
+
+### Discover Casting Players
+
+_{Complete Discovery examples:
+[Android](android/App/app/src/main/java/com/matter/casting/DiscoveryExample.java)
+| [iOS](darwin/TvCasting/TvCasting/MTRDiscoveryExampleViewModel.swift)}_
+
+The Casting Client discovers `CastingPlayers` using Matter Commissioner
+discovery over DNS-SD by listening for CastingPlayer events as they are
+discovered, updated, or lost from the network.
+
+On Android, define a concrete anonymous `class castingPlayerChangeListener` that
+implements the `CastingPlayerChangeListener`.
+
+```java
+private static final CastingPlayerChangeListener castingPlayerChangeListener = new CastingPlayerChangeListener()
+{
+    private final String TAG = CastingPlayerChangeListener.class.getSimpleName();
+
+    @Override
+    public void onChanged(List<CastingPlayer> castingPlayers) {
+        Log.i(TAG, "Discovered changes to " + castingPlayers.size() + " CastingPlayers");
+        // consume changes to the provided castingPlayers
+    }
+
+    @Override
+    public void onAdded(List<CastingPlayer> castingPlayers) {
+        Log.i(TAG, "Discovered " + castingPlayers.size() + " CastingPlayers");
+        // consume discovered castingPlayers
+    }
+
+    @Override
+    public void onRemoved(List<CastingPlayer> castingPlayers) {
+        Log.i(TAG, "Removed " + castingPlayers.size() + " CastingPlayers");
+        // consume castingPlayers removed or lost from the network
+    }
+};
+```
+
+On iOS, implement a `func addDiscoveredCastingPlayers`,
+`func removeDiscoveredCastingPlayers` and `func updateDiscoveredCastingPlayers`
+which listen to notifications as Casting Players are added, removed, or updated.
+
+```objectivec
+@objc
+func addDiscoveredCastingPlayers(notification: Notification)
```

Comment:
The function names do not really align with them being a handle of a notification. Maybe something like didDiscoverCastingPlayers? The same applies for the other functions.

Reply:
Done